persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/MushroomDaoTest.java

```diff
@@ -0,0 +1,260 @@
+        assertThatThrownBy(() -> mushroomDao.findById(null)).hasRootCauseExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void create() throws Exception {
+        Mushroom mushroom = createMushroom("WierdMushroom", MushroomType.UNEDIBLE, "May", "July");
+        mushroomDao.create(mushroom);
+
+        List<Mushroom> list = em.createQuery("select m from Mushroom m", Mushroom.class)
+                .getResultList();
+
+        assertThat(list).containsExactlyInAnyOrder(mushroom1,mushroom2,mushroom);
+
+    }
+
+    @Test
+    public void create_nullName() throws Exception {
+        Mushroom mushroom = createMushroom(null, MushroomType.UNEDIBLE, "May", "July");
+
+        assertThatThrownBy(() -> mushroomDao.create(mushroom)).isInstanceOf(ConstraintViolationException.class);
+    }
+
+    @Test
+    public void create_nullMushroomType() throws Exception {
+        Mushroom mushroom = createMushroom("WierdMushroom", null, "May", "July");
+
+        assertThatThrownBy(() -> mushroomDao.create(mushroom)).isInstanceOf(ConstraintViolationException.class);
+    }
+
+    //this string validation is not implemented yet - tests prepared for the future
+//    @Test
+//    public void create_nullBeginMonth() throws Exception {
+//        Mushroom mushroom = createMushroom("WierdMushroom", MushroomType.UNEDIBLE, null, "July");
+//
+//        assertThatThrownBy(() -> mushroomDao.create(mushroom)).isInstanceOf(ConstraintViolationException.class);
+//    }
+//
+//    @Test
+//    public void create_nullLastMonth() throws Exception {
+//        Mushroom mushroom = createMushroom("WierdMushroom", MushroomType.UNEDIBLE, "June", null);
+//
+//        assertThatThrownBy(() -> mushroomDao.create(mushroom)).isInstanceOf(ConstraintViolationException.class);
+//    }
+
+    @Test
+    public void create_nullEntity() throws Exception {
+
+        assertThatThrownBy(() -> mushroomDao.create(null)).hasRootCauseExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void delete() throws Exception {
+
+        mushroomDao.delete(mushroom1);
+
+        List<Mushroom> list  = em.createQuery("select m from Mushroom m", Mushroom.class).getResultList();
+
+        assertThat(list).hasSize(1);
+
+    }
+
+    @Test
+    public void delete_entityNotPersisted() throws Exception {
+        Mushroom mush = createMushroom("mush", MushroomType.UNEDIBLE, "May","June");
+        mushroomDao.delete(mush); // should not do anything
+        em.flush();
+        assertThat(mushroomDao.findAll()).containsExactlyInAnyOrder(mushroom1, mushroom2);
+    }
+
+    @Test
+    public void delete_nullEntity() throws Exception {
+        assertThatThrownBy(() -> mushroomDao.delete(null)).hasRootCauseInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findAll() throws Exception {
+
+        List<Mushroom> list = mushroomDao.findAll();
+
+        assertThat(list).hasSize(2);
```

Comment:
feel free to check the contents of the list too (containsExactlyInAnyOrder(...))